packages/web-api/src/controllers/scan-result-controller.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Context } from '@azure/functions';
+import { GuidGenerator, ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import { Logger } from 'logger';
+import { OnDemandPageScanRunResultProvider } from 'service-library';
+import { ItemType, OnDemandPageScanResult } from 'storage-documents';
+
+import { Dictionary, isEmpty, keyBy } from 'lodash';
+import { webApiIocTypes } from '../setup-ioc-container';
+import { ApiController } from './api-controller';
+
+@injectable()
+export class ScanResultController extends ApiController {
+    public readonly apiVersion = '1.0';
+    public readonly apiName = 'get-scan';
+
+    public constructor(
+        @inject(webApiIocTypes.azureFunctionContext) protected readonly context: Context,
+        @inject(OnDemandPageScanRunResultProvider) private readonly onDemandPageScanRunResultProvider: OnDemandPageScanRunResultProvider,
+        @inject(GuidGenerator) private readonly guidGenerator: GuidGenerator,
+        @inject(ServiceConfiguration) protected readonly serviceConfig: ServiceConfiguration,
+        @inject(Logger) protected readonly logger: Logger,
+    ) {
+        super();
+    }
+
+    public async handleRequest(): Promise<void> {
+        const scanId = <string>this.context.bindingData.scanId;
+        const timeRequested = this.tryGetScanRequestedTime(scanId);
+        if (timeRequested === undefined) {
+            // the scanId is invalid.
+            return;
+        }
+
+        const timeCurrent = new Date();
+        const scanResultQueryBufferInSeconds = (await this.getRestApiConfig()).scanResultQueryBufferInSeconds;
```

Comment:
scanResultQueryBufferInSeconds  => this is not a buffer. rename to minimumWaitTimeforScanStatusCall